### ISSUE TITLE
Refactor how dependencies work in SakuraApi

### ISSUE THREAD

**Issue by etsuo** [sakuraapi/api#83]
history...

- started with singleton of SakuraApi... that was a mistake...
- refactored to inject SakuraApi into `@Routable` and `@Model`... that was progress
- that lead, however, to testing problem when actually integrating since the project only had a "production" instance of sapi and it wasn't possible to swap it out for a test version... basically it was global state that wasn't easily overridden for testing... <sigh>
- enter the hack... `.changeSapi` introduced on `@Model` and `@Routable`... got me past the first round of testing when integrating SakuraApi with a project that Olive Technology is working on... 
- but then more problems... 

Regrets: Yes, I should have gone with a dependency injection approach from the get go... time to refactor.

Goal: make SakuraApi MUCH more testable by using some sort of dependency injection based design pattern.

**Comment by etsuo** [comment]
#80 is probably a symptom of the problems raised with how the SakuraApi object is implemented.

**Comment by etsuo** [comment]
#81 is probably a symptom of the issues this ticket hopes to address.

**Comment by etsuo** [comment]
Commit 1a396e3 was me even though it's not verified... I updated with bbb1289, which is properly signed.

**Comment by etsuo** [comment]
SakuraApi now requires explicit dependency injection when SakuraApi is instantiated:
```
const sapi = new SakuraApi({
    models: [SomeModel],
    routables: [SomeRoutable]
  });
```

It also now supports the notion of modules, which are injectable addons like auth-native-authority. For Example:
```
const sapi = new SakuraApi({
    models: di.models,
    routables: di.routables,
    modules: [
      {module: addAuthenticationAuthority, options: authOptions}
    ],
    configPath: 'dist/spec/config/environment.json'
  });
  sapi.baseUri = baseUri;```

**Comment by etsuo** [comment]
Lint wasn't working, that got fixed... along with all the overlooked lints.

**Comment by etsuo** [comment]
Released on 0.7.0-3
